jetstream/argo.py

```diff
@@ -89,7 +89,20 @@ def submit_workflow(
                     failed = True
 
             time.sleep(1)
-    return not failed
+
+    # check status of pods
+    all_pods_succeeded = True
+
+    if workflow.status and workflow.status.nodes:
+        all_pods_succeeded = all(
+            [
+                node.phase == "Succeeded"
+                for _, node in workflow.status.nodes.items()
+                if node.type == "Pod"
+            ]
+        )
```

Comment:
Just checking the workflow status is unfortunately not enough, we need to check for every pod if it succeeded.
In the workflow config the `analyse-experiment` has `continueOn: failed: true`, so if one of these steps fails, the workflow continues running and ends in a succeeded state.